### PR TITLE
message.pyi: Now takes Address as EmailMessage header

### LINKED ISSUE #2863: Email example from python docs doesn't validate: Address as EmailMessage header
`EmailMessage.__setitem__` should support `Address` as a value. At least the Python docs email example think it does. See https://docs.python.org/3/library/email.examples.html

test_mypy.py:
```
from email.message import EmailMessage
from email.headerregistry import Address

msg = EmailMessage()
msg['From'] = Address("Pepé Le Pew", "pepe", "example.com")
```

And then:
```
$ mypy test_mypy.py
test_mypy.py:5: error: Incompatible types in assignment (expression has type "Address", target has type "Union[str, Header]")
```

### PR DESCRIPTION
**Pull request by utkarsh2102**
Fixes #2863

### CODE REVIEW

**Comment by utkarsh2102** [comment]
@srittau, probably it'd be better to have it accept, set and delete `Any`. The latest commit does that instead of changing the `_HeaderType` to `Any`.  
Let me know if this suffices?

File: stdlib/3/email/message.pyi [hunk 1]
```diff
@@ -34,9 +34,9 @@ class Message:
     def get_charset(self) -> _CharsetType: ...
     def __len__(self) -> int: ...
     def __contains__(self, name: str) -> bool: ...
-    def __getitem__(self, name: str) -> Optional[_HeaderType]: ...
-    def __setitem__(self, name: str, val: _HeaderType) -> None: ...
-    def __delitem__(self, name: str) -> None: ...
+    def __getitem__(self, name: str) -> Any: ...
+    def __setitem__(self, name: str, val: _HeaderType) -> Any: ...
```
Comment (JelleZijlstra) [inline stdlib/3/email/message.pyi]: Why did you change the return types? I think the argument type is the one that needs changing.

Comment (utkarsh2102) [reply stdlib/3/email/message.pyi]: @JelleZijlstra, would it make sense to have the argument type defined like this:
`_HeaderType = Union[str, Header, ContentTransferEncodingHeader, ContentTypeHeader, UniqueDateHeader, SingleAddressHeader, AddressHeader, UniqueAddressHeader, UniqueSingleAddressHeader, MIMEVersionHeader, UniqueUnstructuredHeader, DateHeader, ContentDispositionHeader]`?

Comment (JelleZijlstra) [reply stdlib/3/email/message.pyi]: That's way too big a union to be a good idea.

Comment (utkarsh2102) [reply stdlib/3/email/message.pyi]: Right on, what's the best approach then?

Comment (utkarsh2102) [reply stdlib/3/email/message.pyi]: Hi @JelleZijlstra.  
Whilst going over the pending tasks, I stumbled upon this.  
What'd be the best approach here?
CC: @srittau.

Comment (srittau) [reply stdlib/3/email/message.pyi]: I think the best way forward is to change `_HeaderType` to `Any` due to the complicated type situation and the fact it is used in return types. The `__getitem__()` should return `_HeaderType`. `__setitem__()` and `__delitem__()` should continue to return `None`, though.

Comment (utkarsh2102) [reply stdlib/3/email/message.pyi]: Thank you, fixed!
